### PR TITLE
add LoadingSnackbar and use it in sagas, update notistack

### PR DESCRIPTION
**Pull request by p6te**
- Create LoadingSnackbar component,
- update notistack library for easier customization of snackbars,
- use loadingSnackbar in sagas
- fix run command for storybook

image

image

### CODE REVIEW

**Review by wojciech-cichocki — COMMENTED** [review]
signing transaction should disappear after the transaction is signed, not after receiving a response from the rpc:
image

**Review by wojciech-cichocki — CHANGES_REQUESTED** [review]
Also removed details from the snackbar, such as the amount of tokens

**Review by wojciech-cichocki — CHANGES_REQUESTED** [review]
"When I want to close the position (SOL/USDC), I get a loader for claiming fees and closing the position. The valid solution is only closing the position.

(It is also possible that the closing position saga includes claiming fees. If so, you should remove the claiming fee, as closing the position already includes the feature of claiming fees.)

**Comment by wojciech-cichocki** [comment]
> "When I want to close the position (SOL/USDC), I get a loader for claiming fees and closing the position. The valid solution is only closing the position.
> 
> (It is also possible that the closing position saga includes claiming fees. If so, you should remove the claiming fee, as closing the position already includes the feature of claiming fees.)

Screenshot from 2024-03-23 22-45-28

File: src/store/sagas/bonds.ts [hunk 1]
```diff
@@ -84,7 +85,19 @@ export function* handleGetUserVested() {
 }
 
 export function* handleBuyBondWithWSOL(data: BuyBond) {
```
Comment (wojciech-cichocki) [inline src/store/sagas/bonds.ts:87]: Just for your information: currently, bounds are not used in production. They probably won't be used in the near future either. The method of fundraising in crypto has changed, so there is no need to maintain this feature as it was. However, if it has already been updated, that's good.

File: src/store/sagas/positions.ts [hunk 2]
```diff
@@ -47,8 +48,19 @@ function* handleInitPositionAndPoolWithSOL(action: PayloadAction<InitPositionDat
   ) {
     return yield* call(handleInitPosition, action)
   }
-
+  const loaderCreatePool = (new Date().getMilliseconds() + Math.random()).toString()
+  const loaderSigningTx = (new Date().getMilliseconds() + Math.random()).toString()
```
Comment (wojciech-cichocki) [inline src/store/sagas/positions.ts:52]: The method of determining the `key` can also be encapsulated in a utility function, making it easier to control it in one place. This formula, although short, is repeated multiple times

**Review by wojciech-cichocki — CHANGES_REQUESTED** [review]

File: src/store/sagas/positions.ts (hunk 2 above)
Comment (p6te) [reply src/store/sagas/positions.ts]: that's true, i've created createLoaderKey util for it.

File: src/components/Snackbar/LoadingSnackbar/style.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { colors, typography } from '@static/theme'
+import { CircularProgress, styled, Typography } from '@material-ui/core'
+import { SnackbarContent } from 'notistack'
+
+export const StyledSnackbarContent = styled(SnackbarContent)(({ theme }) => ({
```
[the remark this answers is not in the document]
Comment (p6te) [reply src/components/Snackbar/LoadingSnackbar/style.ts:5]: yes, now it's adjusted. I have used a smaller font for mobile. we don't have an example of it on mockups.

File: src/components/Snackbar/LoadingSnackbar/LoadingSnackbar.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import { CustomContentProps, useSnackbar } from 'notistack'
```
[the remark this answers is not in the document]
Comment (p6te) [reply src/components/Snackbar/LoadingSnackbar/LoadingSnackbar.tsx:1]: yes, i've added this component to storybook

**Review by wojciech-cichocki — APPROVED** [review]

**Pull request merged**